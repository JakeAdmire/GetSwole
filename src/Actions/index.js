export const addUser = (name) => ({
  type: 'ADD_USER',
  name
})

export const addDateToStore = (date) => ({
  type: 'ADD_DATE_TO_STORE',
  date
})

export const addExercises = (exercises) => ({
  type: 'ADD_EXERCISES',
  exercises
})