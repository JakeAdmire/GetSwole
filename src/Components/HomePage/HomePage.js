import React from 'react';
import { StyleSheet, Text, View, BackHandler } from 'react-native';
import { Header } from 'react-native-elements';
import { connect } from 'react-redux';
// 
import { addExercises } from '../../Actions';
import Calendar from '../Calendar/Calendar';
import Dimensions from 'Dimensions';
import Container from '../Container/Container';
import { RalewayText, RalewayBoldText } from '../../Utilities/RalewayText';
import * as palette from '../../Utilities/styleIndex';

export class Homepage extends React.Component {
  static navigationOptions = {
    header: null
  }

  componentDidMount() {
    this.handleAddExercises()
  }

  handleAddExercises = async () => {
    const response = await fetch('https://warm-cove-89223.herokuapp.com/api/v1/exercises');
    const data = await response.json();
    this.props.addExercises(data)
  }

  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackButton);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackButton);
  }

  handleBackButton = () => {
    BackHandler.exitApp();
    return true;
  };

  welcomeText = () => {
    return  <Text style={styles.headerText}>Welcome, 
              <Text style={{ fontFamily: 'raleway-bold', color: palette.darkAccent }}> {this.props.user.name}</Text>
            </Text>
  }

  render() {
    const { navigation, date } = this.props;
    return (
      <View style={styles.HomePage}>

        <Header centerComponent={this.welcomeText}
                containerStyle={{ backgroundColor: '#FFF', elevation: 12 }} />

        <View style={styles.prompt}>
          <Text style={styles.promptText}>Select a day to start scheduling workouts</Text>
        </View>

        <Calendar />
        { 
          date 
            ? <View>
                <Container navTool={navigation} />
              </View>
            : <View style={styles.placeholder}>
              <Text style={styles.placeholderText}>Please select a date</Text>
            </View>
        }
      </View>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  date: state.date
});

export const mapDispatchToProps = (dispatch) => ({
  addExercises: (exercises) => dispatch(addExercises(exercises))
});

export default connect(mapStateToProps, mapDispatchToProps)(Homepage);

const styles = StyleSheet.create({

  HomePage: {
    // backgroundColor: '#667D90',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height
  },

  header: {
    position: 'relative',
    top: -10,

    backgroundColor: '#FFF',
    height: 80,
    paddingTop: 30,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
    elevation: 12
  },

  headerText: {
    fontFamily: 'raleway',
    fontSize: 24,
    color: palette.backgroundColor,
  },

  prompt: {
    height: 100,
    backgroundColor: palette.backgroundColor,
    justifyContent: 'center',
    elevation: 11
  },

  promptText: {
    color: palette.lightAccent,
    fontSize: 20,
    padding: 10,
  },

  placeholder: {
    backgroundColor: palette.backgroundColor,
    height: Dimensions.get('window').height - 185,
    padding: 30,
    alignItems: 'center'
  },

  placeholderText: {
    fontFamily: 'raleway-bold',
    color: palette.lightAccent,
    fontSize: 30
  }

});
