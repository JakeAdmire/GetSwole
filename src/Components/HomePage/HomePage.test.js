<<<<<<< HEAD
import React from 'react'
import { shallow } from 'enzyme'
import { Homepage, mapStateToProps, mapDispatchToProps } from './Homepage'
import { addExercises } from '../../Actions'

describe('Homepage', () => {

  let wrapper;
  let mockDate = '2019 5 25';
  let semanticDateMock = '25th March, 2019';
  let loadingMock = false;
  let mockUser = { name: 'John' }
  let routinesMock = {
    data: [
      {
        id: "4",
        type: "routine",
        attributes: {
          name: "Arm Day",
          exercises: [
            {
              id: 50,
              name: "Pushups",
              category: "Strength",
              equipment_required: "Body Only",
              muscle: "Chest",
              reps: 10,
              sets: 4,
              duration: null,
              weight: null
            },
            {
              id: 70,
              name: "Chin-Up",
              category: "Strength",
              equipment_required: "Body Only",
              muscle: "Lats",
              reps: 5,
              sets: 4,
              duration: null,
              weight: null
            },
            {
              id: 104,
              name: "Barbell Curl",
              category: "Strength",
              equipment_required: "Barbell",
              muscle: "Biceps",
              reps: 10,
              sets: 4,
              duration: null,
              weight: null
            }
          ]
        }
      }
    ]
  };
  beforeEach(() => {
    wrapper = shallow(<Homepage />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
    let date = { date: '2019 05 29' }
    wrapper.setProps({ date })
    expect(wrapper).toMatchSnapshot()
  })


  describe('mapDispatchToProps', () => {
    it('should dispatch ADD EXERCISES if the props function is triggered', () => {
      const mockDispatch = jest.fn()
      const mockExercises = {
        exercises: [
          {
            id: 50,
            name: "Pushups",
            category: "Strength",
            equipment_required: "Body Only",
            muscle: "Chest",
            reps: 10,
            sets: 4,
            duration: null,
            weight: null
          }
        ]
      }
      const actionToDispatch = addExercises(mockExercises)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addExercises(mockExercises)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
  describe.skip('mapStateToProps', () => {
    it('should map State to Props', () => {
      let mockState = {
        name: 'Helen',
        date: mockDate,
        semanticDate: semanticDateMock,
        routines: routinesMock,
        loading: loadingMock
      };
      let mockUpdatedState = {
        name: 'Helen',
        date: mockDate,
      }
      const results = mapStateToProps(mockState)
      expect(results).toEqual(mockUpdatedState)
    })
  })

  describe('componentDidMount', () => {
    it('should call handleAddExercises', () => {
      const instance = wrapper.instance()
      jest.spyOn(instance, 'handleAddExercises')
      instance.componentDidMount()
      expect(instance.handleAddExercises).toHaveBeenCalled()
    })
  })
  
})
=======
import React from 'react';
import { shallow } from 'enzyme';
// 
import { Homepage, mapStateToProps, mapDispatchToProps } from './HomePage';
import { addExercises } from '../../Actions';

describe('HomePage', () => {

  let wrapper;
  // 
  let navigationMock = { navigate: jest.fn() };
  let addExercisesMock = jest.fn();
  let userMock = { id: 1, name: 'Jim' };
  let dateMock = '12 05 1997' // ?

  beforeEach(() => {
    wrapper = shallow(
      <Homepage navigation={navigationMock}
                addExercises={addExercisesMock}
                user={userMock}
                date={dateMock} />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('handleBackButton()', () => {

    it('should return true', () => {
      let results = wrapper.instance().handleBackButton();

      expect(results).toEqual(true);
    })

  })

  describe('welcomeText()', () => {

    it('should match snapshot', () => {
      let results = wrapper.instance().welcomeText();

      expect(results).toMatchSnapshot();
    })

  })

})

describe('mapStateToProps', () => {

  let mockState = {
    user: 'Joel',
    date: '10 29 12'
  }

  it('should return a props object', () => {
    let results = mapStateToProps(mockState);

    let expected = { user: mockState.user, date: mockState.date };

    expect(results).toEqual(expected);
  })

})

describe('mapDispatchToProps', () => {

  it('should dispatch an addExercises method', () => {
    let mockExercises = [];
    const mockDispatch = jest.fn();
    const actionToDispatch = addExercises(mockExercises);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.addExercises(mockExercises);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })

})
>>>>>>> dd8075c0cc4d7abb498ef17e217e0a4b76ee6ce6
