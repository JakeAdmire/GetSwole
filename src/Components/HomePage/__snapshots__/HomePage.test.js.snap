// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
exports[`Homepage should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Homepage should match the snapshot 2`] = `ShallowWrapper {}`;
=======
exports[`HomePage should match snapshot 1`] = `ShallowWrapper {}`;

exports[`HomePage welcomeText() should match snapshot 1`] = `
<Component
  style={
    Object {
      "color": "#2D71A8",
      "fontFamily": "raleway",
      "fontSize": 24,
    }
  }
>
  Welcome,
  <Component
    style={
      Object {
        "color": "#104D80",
        "fontFamily": "raleway-bold",
      }
    }
  >
     
    Jim
  </Component>
</Component>
`;
>>>>>>> dd8075c0cc4d7abb498ef17e217e0a4b76ee6ce6
